kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/BootstrapServersAnnotation.java

```diff
@@ -0,0 +1,109 @@
+ * connect with. We include (clusterName, ingressName, bootstrapServers) for each bootstrap address that the
+ * client should eventually host.
+ * <p>
+ * We also include a version in the value of the annotation, in case we ever need to break compatibility but
+ * continue reading older Service's metadata.
+ * </p>
+ */
+public class BootstrapServersAnnotation {
+
+    public static final String BOOTSTRAP_SERVERS_ANNOTATION_KEY = "kroxylicious.io/bootstrap-servers";
+
+    private BootstrapServersAnnotation() {
+    }
+
+    /**
+     * Adds a `kroxylicious.io/bootstrap-servers`annotation to the supplied metadata fluent
+     * @param meta the Metadata fluent builder to add the annotation to
+     * @param bootstrapServers the bootstrap servers to serialize into the annotation value
+     */
+    public static void annotate(ObjectMetaFluent<?> meta, Set<BootstrapServer> bootstrapServers) {
+        if (bootstrapServers.isEmpty()) {
+            return;
+        }
+        meta.addToAnnotations(BOOTSTRAP_SERVERS_ANNOTATION_KEY, toAnnotation(bootstrapServers));
+    }
+
+    /**
+     * Read bootstrap servers from ObjectMeta, extracting them from an `kroxylicious.io/bootstrap-servers`
+     * annotation if present.
+     * @param meta the metadata to extract the bootstrap servers from
+     * @return the bootstrap servers from the metadata if the annotation is present, else an empty Set
+     */
+    public static Set<BootstrapServer> bootstrapServersFrom(ObjectMeta meta) {
+        if (!meta.getAnnotations().containsKey(BOOTSTRAP_SERVERS_ANNOTATION_KEY)) {
+            return Set.of();
+        }
+        else {
+            return fromAnnotation(meta.getAnnotations().get(BOOTSTRAP_SERVERS_ANNOTATION_KEY));
+        }
+    }
+
+    @JsonPropertyOrder({ "clusterName", "ingressName", "bootstrapServers" })
+    public record BootstrapServer(String clusterName, String ingressName, String bootstrapServers) {
+        public BootstrapServer {
+            Objects.requireNonNull(clusterName);
+            Objects.requireNonNull(ingressName);
+            Objects.requireNonNull(bootstrapServers);
+        }
+    }
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @VisibleForTesting
+    @JsonPropertyOrder({ "version", "bootstrapServers" })
+    record Wrapper(String version, List<BootstrapServer> bootstrapServers) {
+        Wrapper {
+            Objects.requireNonNull(version);
+            Objects.requireNonNull(bootstrapServers);
+        }
+    }
+
+    private static String toAnnotation(Set<BootstrapServer> bootstrapServers) {
+        List<BootstrapServer> list = bootstrapServers.stream()
+                .sorted(Comparator.comparing(BootstrapServer::clusterName).thenComparing(BootstrapServer::ingressName).thenComparing(BootstrapServer::bootstrapServers))
+                .toList();
+        Wrapper wrapper = new Wrapper("1", list);
```

Comment:
Added this Wrapper with version just in case we want to break compatibility but keep reading old Services on initial deploy of a new operator version. I guess we could alternatively use the annotation name and do it that way, and omit the Wrapper.

Reply:
yep, sounds good